libs/wingsdk/src/target-sim/event-mapping.ts

```diff
@@ -0,0 +1,82 @@
+import { Construct } from "constructs";
+import { ISimulatorResource } from "./resource";
+import { BaseResourceSchema } from "./schema";
+import {
+  EVENT_MAPPING_TYPE,
+  EventMappingSchema,
+  EventSubscription,
+} from "./schema-resources";
+import { simulatorHandleToken } from "./tokens";
+import { bindSimulatorResource, makeSimulatorJsClient } from "./util";
+import { fqnForType } from "../constants";
+import * as core from "../core";
+import { Resource, App } from "../core";
+
+export const EVENT_MAP_FQN = fqnForType("sim.EventMapping");
+
+export interface EventProps {}
+
+export abstract class EventMappingBase extends Resource {
+  public static _newEventMapping(
+    scope: Construct,
+    id: string,
+    props: EventProps = {}
+  ): Event {
+    return App.of(scope).newAbstract(EVENT_MAP_FQN, scope, id, props);
+  }
+  public readonly stateful = true;
+
+  constructor(scope: Construct, id: string, props: EventProps = {}) {
+    super(scope, id);
+    props;
+  }
+}
+
+export interface IEventPublisher {
+  addEventSubscription: (payload: any) => Promise<void>;
+}
```

Comment:
Let's make sure we convey somehow that this needs to be implemented by inflight clients, not preflight classes, eg
```suggestion
export interface IEventPublisher extends ISimulatorResourceInstance {
  addEventSubscription: (payload: any) => Promise<void>;
}
```